gomock/argument_captor.go

```diff
@@ -0,0 +1,37 @@
+package gomock
+
+type ArgumentCaptor interface {
```

Comment:
There is no need to export the ArgumentCaptor interface. Reasons:
1. Exported interface means we never can add or remove methods from it without breaking backwards compatibilty
2. Even in the this PR, there is no practical reason for the interface. It has only one implementation

Instead I would argue just to export the `ArgumentCaptor` struct and make `Captor()` method return `*ArgumentCaptor`